### ISSUE TITLE
ParameterAssignment does not detect problem for Lambda parameters

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#11038]
I have read check documentation: https://checkstyle.org/config_coding.html#ParameterAssignment
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="ParameterAssignment"/>
  </module>
</module>

$ cat Test.java 
import java.util.*;
public class Test {
    public void foo() {
        ArrayList<Integer> list = new ArrayList<>();
        list.add(1);
        list.add(2);
        list.forEach((i) -> {
            i *= 10;  // should report a warning
        });
    }
}

$ java $RUN_LOCALE -jar checkstyle-9.2-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

violation is expected at ` i *= 10;  // should report a warning` line

**Comment by strkkk** [comment]
I do not think we need a new check for this case. It is not something special, I doubt anybody needs to assign something to lambda parameter. So I would say it should behave in same way as for parameters

**Comment by romani** [comment]
```
java $RUN_LOCALE -jar checkstyle-9.2-all.jar -t Test.java | awk '/LAMBDA/,/RCURLY/' | sed "s/        |       |       |   //g"
`--LAMBDA -> -> [7:25]
    |--LPAREN -> ( [7:21]
    |--PARAMETERS -> PARAMETERS [7:22]
    |   `--PARAMETER_DEF -> PARAMETER_DEF [7:22]
    |       |--MODIFIERS -> MODIFIERS [7:22]
    |       |--TYPE -> TYPE [7:22]
    |       `--IDENT -> i [7:22]
    |--RPAREN -> ) [7:23]
    `--SLIST -> { [7:28]
        |--EXPR -> EXPR [8:14]
        |   `--STAR_ASSIGN -> *= [8:14]
        |       |--IDENT -> i [8:12]
        |       `--NUM_INT -> 10 [8:17]
        |--SEMI -> ; [8:19]
        `--RCURLY -> } [9:8]
```

**Comment by nmancus1** [comment]
> I do not think we need a new check for this case.

I agree.

We should also consider case like:

```java

        list.forEach(i -> i *= 10);

```

Note lack of `PARAMETERS`/`PARAMETER_DEF` node:

```
        |       |   `--METHOD_CALL -> ( [8:20]
        |       |       |--DOT -> . [8:12]
        |       |       |   |--IDENT -> list [8:8]
        |       |       |   `--IDENT -> forEach [8:13]
        |       |       |--ELIST -> ELIST [8:23]
        |       |       |   `--LAMBDA -> -> [8:23]
        |       |       |       |--IDENT -> i [8:21]
        |       |       |       `--EXPR -> EXPR [8:28]
        |       |       |           `--STAR_ASSIGN -> *= [8:28]
        |       |       |               |--IDENT -> i [8:26]
        |       |       |               `--NUM_INT -> 10 [8:31]
        |       |       `--RPAREN -> ) [8:33]
```

**Comment by Vyom-Yadav** [comment]
I am on it, I have some PR's open but all the work from my side in those PR's is up to date. :)
